gems/pending/VMwareWebService/MiqPbmInventory.rb

```diff
@@ -50,20 +50,21 @@ def pbmQueryAssociatedEntity(profile_ids)
     assoc_entities
   end
 
-  def pbmQueryMatchingHub(profile_id)
-    hubs = []
+  def pbmQueryMatchingHub(profile_ids)
+    hubs = {}
     return hubs if @pbm_svc.nil?
 
     begin
-      # If a string was passed in create a PbmProfileId object
-      profile_id = RbVmomi::PBM::PbmProfileId(:uniqueId => profile_id) if profile_id.kind_of?(String)
+      profile_ids.each do |profile_id|
+        # If a string was passed in create a PbmProfileId object
+        profile_id = RbVmomi::PBM::PbmProfileId(:uniqueId => profile_id) if profile_id.kind_of?(String)
```

Comment:
@Fryguy initializing `MiqPbmInventory` is done right after `MiqVimInventory` in `MiqVim#initialize` so it is accessing the session cookie under the connection lock in MiqVimBroker#getMiqVim`, however it is storing it by reference not by value so subsequent calls access those values outside the connection lock.  I'll put in a separate PR to dup the session cookie while under the connection lock.